### DISCUSSION TITLE
Trouble building android on 2024.07 and master

### DISCUSSION THREAD

**Discussion by hasezoey** [koreader/koreader#12346] [Q&A] [answered]
I am having some trouble building the android targets with `2024.07` and current master(`d706a129fa14dadc4ba19547442989430f119e33`), both with different errors.

2024.07 fails with that it cannot find the Java Compiler:

```txt
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:compileArmRocksReleaseJavaWithJavac'.
> Java compiler is not available. Please check that /usr/lib/jvm/java-17-openjdk contains a valid JDK installation.
```

current master fails with:
```txt
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:mergeArmRocksReleaseAssets'.
> [module/version.txt] /mnt/projects/android/koreader/platform/android/luajit-launcher/assets/module/version.txt        [module/version.txt] /mnt/projects/android/koreader/koreader-android-armv7a-unknown-linux-android18/luajit-launcher/assets/module/version.txt: Error: Duplicate resources
  [module/map.txt] /mnt/projects/android/koreader/platform/android/luajit-launcher/assets/module/map.txt        [module/map.txt] /mnt/projects/android/koreader/koreader-android-armv7a-unknown-linux-android18/luajit-launcher/assets/module/map.txt: Error: Duplicate resources
  [module/koreader.7z] /mnt/projects/android/koreader/platform/android/luajit-launcher/assets/module/koreader.7z        [module/koreader.7z] /mnt/projects/android/koreader/koreader-android-armv7a-unknown-linux-android18/luajit-launcher/assets/module/koreader.7z: Error: Duplicate resources
```
(already tried removing everything in `/mnt/projects/android/koreader/koreader-android-armv7a-unknown-linux-android18`)

and i dont know enough about the build system to fix any of the 2 issues

---

i did some bisecting, i started with narrowing down koreader itself starting at `e1f2acf68f00d1c02c4dac070ef12b2b59be49bd` as that is the last to me known good commit.
the result here is that [`b7fc048bbe5514cdc2c8ddd122503ef80f9543fc`](https://github.com/koreader/koreader/commit/b7fc048bbe5514cdc2c8ddd122503ef80f9543fc) is the first broken commit.

but seeing that bad commit updated `android-luajit-launcher`, i did some bisecting there, starting at `987f37ce34388d75e35f09227c0230b3f226c6f8` (the value it was before `b7fc048bbe5514cdc2c8ddd122503ef80f9543fc`).
the result here is that [`a647e7b54aa17398dbb22b2bc7f2e01ed8cc461c`](https://github.com/koreader/android-luajit-launcher/commit/a647e7b54aa17398dbb22b2bc7f2e01ed8cc461c) is the first failing commit with the compiler not available error

---

forgot to mention, my system is manjaro 24.0.6 trying to use `extra/jdk17-openjdk`

**Comment by benoit-pierre** [comment]
I don't have any issue on Arch Linux.

Since bd4a651fbf, the luajit-launcher build is done out-of-tree, can you check the `platform/android/luajit-launcher` checkout is clean? The only thing created should be an empty `.gradle/kotlin/errors` directory.

Additionally, is everything else (NDK, SDK, build-tools, etc…) correct?

**Reply by hasezoey** [reply]
>  can you check the platform/android/luajit-launcher checkout is clean?

the checkout is clean, i only modified `gradlew` to add `--info` for extra information (same error if completely clean)

> Additionally, is everything else (NDK, SDK, build-tools, etc…) correct?

what exactly is meant with correct? it is the same as in the working commits as android ndk should not need updating right? also which sdk?

aside from that, i also used `./kodev clean` and `./kodev clean android`.

**Reply by benoit-pierre** [reply]
The duplicate resources point to the luajit-launcher checkout not being clean (if you built previous to bd4a651, you need to manually remove old build files: `git -C platform/android/luajit-launcher clean -x -d -i`).

As for NDK, SDK, etc, see [base/toolchain/Makefile](https://github.com/koreader/koreader-base/blob/master/toolchain/Makefile).

Note that there's also a docker image for Android builds will all the necessary tools: `koreader/koandroid:0.8.3-20.04`.

**Reply by hasezoey** [reply]
i am now back on master (`d706a129fa14dadc4ba19547442989430f119e33`) and did a `git clean -xfd .` in `platform/android/luajit-launcher` which resolved the `duplicate resource` error, but now i am back to the `Java compiler is not available` error.
(note android-luajit-launcher is at `981e9c871ad2b94c58b1f4b01f431bed3d80382d` while being in `d706a129fa14dadc4ba19547442989430f119e33` koreader)

as for the things in the makefile, i have not touched it as it does not look changed since the good commits, and if it did change (`build-tools;34.0.0`) should automatically be changed by kodev right?

**Reply by benoit-pierre** [reply]
> (note android-luajit-launcher is at 981e9c871ad2b94c58b1f4b01f431bed3d80382d while being in d706a129fa14dadc4ba19547442989430f119e33 koreader)

I don't have a `d706a129fa14dadc4ba19547442989430f119e33` revision in my checkout of android-luajit-launcher… Koreader's master (6eddc665166261163276b615ff5aaabf406dc4de) has the checkout as `981e9c871ad2b94c58b1f4b01f431bed3d80382d`. Are you submodules in sync? (You can use `./kodev fetch fetch-thirdparty` to synchronize them).

> as for the things in the makefile, i have not touched it as it does not look changed since the good commits, and if it did change (build-tools;34.0.0) should automatically be changed by kodev right? 

Do you use your own NDK / SDK installs? Are any of the following environment variables set: `ANDROID_HOME`, `ANDROID_SDK_ROOT`, `ANDROID_NDK_HOME` `ANDROID_NDK_ROOT`?

**Reply by hasezoey** [reply]
i am on koreader master `6eddc665166261163276b615ff5aaabf406dc4de` which has `android-luajit-launcher` at `981e9c871ad2b94c58b1f4b01f431bed3d80382d`.

sorry if it confused you but `d706a129fa14dadc4ba19547442989430f119e33` is part of the root koreader tree

> Do you use your own NDK / SDK installs?

thanks for pointing this out, yes `ANDROID_HOME` and `ANDROID_SDK_ROOT` are set to a custom location (the NDK ones are unset), i had that as i needed some android tools outside of koreader and it never seemed to interfere.
(note that i do not have a custom NDK)

i now have tried to unset those variables where kodev then downloaded the new cmdlinetools and updated the own-managed sdk & ndk, but the same `Java compiler is not available. Please check that /usr/lib/jvm/java-17-openjdk contains a valid JDK installation` still persists.

PS: i have now updated the global(custom) SDK tools too, also results in the same error

**Reply by benoit-pierre** [reply]
I don't know… I have installed locally:
```
java-environment-common 3-5
java-runtime-common 3-5
jdk17-openjdk 17.0.12.u7-1
```
Is the default java command using JDK 17?
```
▹ java --version
openjdk 17.0.12 2024-07-16
OpenJDK Runtime Environment (build 17.0.12+7)
OpenJDK 64-Bit Server VM (build 17.0.12+7, mixed mode, sharing)
```

**Reply by hasezoey** [reply]
```sh
$ archlinux-java status             
Available Java environments:
  java-11-openjdk
  java-17-openjdk (default)

$ ll /usr/bin/java
lrwxrwxrwx 1 root root 37 Jan 19  2023 /usr/bin/java -> /usr/lib/jvm/default-runtime/bin/java

$ ll /usr/lib/jvm/default         
lrwxrwxrwx 1 root root 15 Aug 13 15:05 /usr/lib/jvm/default -> java-17-openjdk

$ ll /usr/lib/jvm/default-runtime 
lrwxrwxrwx 1 root root 15 Aug 13 15:05 /usr/lib/jvm/default-runtime -> java-17-openjdk

$ java --version
openjdk 17.0.12 2024-07-16
OpenJDK Runtime Environment (build 17.0.12+7)
OpenJDK 64-Bit Server VM (build 17.0.12+7, mixed mode, sharing)

$ gradle --version

------------------------------------------------------------
Gradle 8.9
------------------------------------------------------------

Build time:    2024-07-11 20:39:33 UTC
Revision:      <unknown>

Kotlin:        1.9.23
Groovy:        3.0.21
Ant:           Apache Ant(TM) version 1.10.13 compiled on January 4 2023
Launcher JVM:  17.0.12 (Arch Linux 17.0.12+7)
Daemon JVM:    /usr/lib/jvm/java-17-openjdk (no JDK specified, using current Java home)
OS:            Linux 6.10.3-1-MANJARO amd64
```

PS: `JAVA_HOME` is also not set

**Reply by benoit-pierre** [reply]
I only have `java-17-openjdk` installed, maybe the `java-11-openjdk` install is causing an issue?

android-luajit-launcher uses its gradlew wrapper for gradle, so the version installed should not matter (I don't have the package installed).

**Reply by hasezoey** [reply]
just uninstalled the jdk11, same error as before

**Reply by benoit-pierre** [reply]
I would:
- unset `ANDROID_HOME`, `ANDROID_SDK_ROOT`, `ANDROID_NDK_HOME` `ANDROID_NDK_ROOT`
- clean existing SDK / NDK directories in `base/toolchain`
- and try `./kodev release -i android` again

**Reply by hasezoey** [reply, marked as answer]
well this is weird, i have found the solution: kill all residual java processes.

i have no clue how they happened, but they are all koreader's processes

<details>
<summary>ps -aux before killing all things</summary>

```txt
ps -aux | grep java                                                  11s
hasezoey  244255  1.2  9.8 10021036 1604476 ?    Ssl  13:33   4:05 /usr/lib/jvm/java-17-openjdk/bin/java --add-opens=java.base/java.util=ALL-UNNAMED --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.lang.invoke=ALL-UNNAMED --add-opens=java.prefs/java.util.prefs=ALL-UNNAMED --add-opens=java.base/java.nio.charset=ALL-UNNAMED --add-opens=java.base/java.net=ALL-UNNAMED --add-opens=java.base/java.util.concurrent.atomic=ALL-UNNAMED -XX:MaxMetaspaceSize=384m -XX:+HeapDumpOnOutOfMemoryError -Xms256m -Xmx512m -Dfile.encoding=UTF-8 -Duser.country=US -Duser.language=en -Duser.variant -cp /home/hasezoey/.gradle/wrapper/dists/gradle-8.6-bin/afr5mpiioh2wthjmwnkmdsd5w/gradle-8.6/lib/gradle-launcher-8.6.jar -javaagent:/home/hasezoey/.gradle/wrapper/dists/gradle-8.6-bin/afr5mpiioh2wthjmwnkmdsd5w/gradle-8.6/lib/agents/gradle-instrumentation-agent-8.6.jar org.gradle.launcher.daemon.bootstrap.GradleDaemon 8.6
hasezoey  246750  0.6  2.7 4942512 447048 ?      Sl   13:38   2:07 /usr/lib/jvm/java-17-openjdk/bin/java -cp /home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-compiler-embeddable/1.9.24/78dab849090e6c5e2eadb6e59a11cdc28fb67a08/kotlin-compiler-embeddable-1.9.24.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib/1.9.24/9928532f12c66ad816a625b3f9984f8368ca6d2b/kotlin-stdlib-1.9.24.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-script-runtime/1.9.24/96771497da90fbc5af1c90fce148db2595a62502/kotlin-script-runtime-1.9.24.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-reflect/1.6.10/1cbe9c92c12a94eea200d23c2bbaedaf3daf5132/kotlin-reflect-1.6.10.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-daemon-embeddable/1.9.24/86685b9c5e8d170925bd9ddbdc263374a6a73281/kotlin-daemon-embeddable-1.9.24.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.intellij.deps/trove4j/1.0.20200330/3afb14d5f9ceb459d724e907a21145e8ff394f02/trove4j-1.0.20200330.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains/annotations/13.0/919f0dfe192fb4e063e7dacadee7f8bb9a2672a9/annotations-13.0.jar -Djava.awt.headless=true -D$java.rmi.server.hostname=127.0.0.1 -Xmx512m -XX:MaxMetaspaceSize=384m -Dkotlin.environment.keepalive -ea --add-exports java.base/sun.nio.ch=ALL-UNNAMED org.jetbrains.kotlin.daemon.KotlinCompileDaemon --daemon-runFilesPath /home/hasezoey/.local/share/kotlin/daemon --daemon-autoshutdownIdleSeconds=7200 --daemon-compilerClasspath /home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-compiler-embeddable/1.9.24/78dab849090e6c5e2eadb6e59a11cdc28fb67a08/kotlin-compiler-embeddable-1.9.24.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib/1.9.24/9928532f12c66ad816a625b3f9984f8368ca6d2b/kotlin-stdlib-1.9.24.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-script-runtime/1.9.24/96771497da90fbc5af1c90fce148db2595a62502/kotlin-script-runtime-1.9.24.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-reflect/1.6.10/1cbe9c92c12a94eea200d23c2bbaedaf3daf5132/kotlin-reflect-1.6.10.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-daemon-embeddable/1.9.24/86685b9c5e8d170925bd9ddbdc263374a6a73281/kotlin-daemon-embeddable-1.9.24.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.intellij.deps/trove4j/1.0.20200330/3afb14d5f9ceb459d724e907a21145e8ff394f02/trove4j-1.0.20200330.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains/annotations/13.0/919f0dfe192fb4e063e7dacadee7f8bb9a2672a9/annotations-13.0.jar
hasezoey  333062  0.0  0.1 8320612 27764 ?       Sl   13:57   0:06 /home/hasezoey/.vscode/extensions/redhat.java-1.33.0-linux-x64/jre/17.0.11-linux-x86_64/bin/java -Dfile.encoding=UTF-8 --add-opens=java.base/java.util=ALL-UNNAMED --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.lang.invoke=ALL-UNNAMED --add-opens=java.prefs/java.util.prefs=ALL-UNNAMED --add-opens=java.base/java.nio.charset=ALL-UNNAMED --add-opens=java.base/java.net=ALL-UNNAMED --add-opens=java.base/java.util.concurrent.atomic=ALL-UNNAMED -classpath /home/hasezoey/.vscode/extensions/vscjava.vscode-gradle-3.16.3/lib/gradle-server.jar com.github.badsyntax.gradle.GradleServer --port=44053 --startBuildServer=true --languageServerPipePath=/run/user/1000/f933a724f78419cbad32835aa9db2ff6.sock --pipeName=/run/user/1000/73ae69bc61682f818082465fc44a7e5d.sock --bundleDir=/home/hasezoey/.vscode/extensions/vscjava.vscode-gradle-3.16.3/server
hasezoey  757055  5.4  5.1 12414940 840808 ?     Ssl  15:43  10:10 /usr/lib/jvm/java-17-openjdk/bin/java --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.lang.invoke=ALL-UNNAMED --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.prefs/java.util.prefs=ALL-UNNAMED --add-opens java.prefs/java.util.prefs=ALL-UNNAMED --add-opens java.base/java.nio.charset=ALL-UNNAMED --add-opens java.base/java.net=ALL-UNNAMED --add-opens java.base/java.util.concurrent.atomic=ALL-UNNAMED -Xmx2048m -Dfile.encoding=UTF-8 -Duser.country=US -Duser.language=en -Duser.variant -cp /home/hasezoey/.gradle/wrapper/dists/gradle-7.5-bin/f1w0cc9nuo0eivz97x2xlu9sv/gradle-7.5/lib/gradle-launcher-7.5.jar org.gradle.launcher.daemon.bootstrap.GradleDaemon 7.5
hasezoey  757584  1.1  2.0 7827872 334116 ?      Sl   15:43   2:04 /usr/lib/jvm/java-17-openjdk/bin/java -cp /home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-compiler-embeddable/1.6.21/97bdd5a752ce73b58ef16c17bc7f0ea2faff7a2e/kotlin-compiler-embeddable-1.6.21.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-reflect/1.6.21/5dc3574d9b7bebfcb4ec6b10ada4aaa9e140bd0b/kotlin-reflect-1.6.21.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib/1.6.21/11ef67f1900634fd951bad28c53ec957fabbe5b8/kotlin-stdlib-1.6.21.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-script-runtime/1.6.21/99c6675100da5d6d5b1c5a1032f27f28008d101b/kotlin-script-runtime-1.6.21.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-daemon-embeddable/1.6.21/46b78cf392e1c4f10854c3cadba44abb7b82250f/kotlin-daemon-embeddable-1.6.21.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.intellij.deps/trove4j/1.0.20200330/3afb14d5f9ceb459d724e907a21145e8ff394f02/trove4j-1.0.20200330.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/net.java.dev.jna/jna/5.6.0/330f2244e9030119ab3030fc3fededc86713d9cc/jna-5.6.0.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib-common/1.6.21/5e5b55c26dbc80372a920aef60eb774b714559b8/kotlin-stdlib-common-1.6.21.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains/annotations/13.0/919f0dfe192fb4e063e7dacadee7f8bb9a2672a9/annotations-13.0.jar -Djava.awt.headless=true -D$java.rmi.server.hostname=127.0.0.1 -Xmx2048m -Dkotlin.environment.keepalive -ea --add-exports java.base/sun.nio.ch=ALL-UNNAMED org.jetbrains.kotlin.daemon.KotlinCompileDaemon --daemon-runFilesPath /home/hasezoey/.local/share/kotlin/daemon --daemon-autoshutdownIdleSeconds=7200 --daemon-compilerClasspath /home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-compiler-embeddable/1.6.21/97bdd5a752ce73b58ef16c17bc7f0ea2faff7a2e/kotlin-compiler-embeddable-1.6.21.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-reflect/1.6.21/5dc3574d9b7bebfcb4ec6b10ada4aaa9e140bd0b/kotlin-reflect-1.6.21.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib/1.6.21/11ef67f1900634fd951bad28c53ec957fabbe5b8/kotlin-stdlib-1.6.21.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-script-runtime/1.6.21/99c6675100da5d6d5b1c5a1032f27f28008d101b/kotlin-script-runtime-1.6.21.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-daemon-embeddable/1.6.21/46b78cf392e1c4f10854c3cadba44abb7b82250f/kotlin-daemon-embeddable-1.6.21.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.intellij.deps/trove4j/1.0.20200330/3afb14d5f9ceb459d724e907a21145e8ff394f02/trove4j-1.0.20200330.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/net.java.dev.jna/jna/5.6.0/330f2244e9030119ab3030fc3fededc86713d9cc/jna-5.6.0.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib-common/1.6.21/5e5b55c26dbc80372a920aef60eb774b714559b8/kotlin-stdlib-common-1.6.21.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains/annotations/13.0/919f0dfe192fb4e063e7dacadee7f8bb9a2672a9/annotations-13.0.jar hasezoey 2208619  0.3  2.4 7343128 395728 ?      Ssl  17:12   0:22 /usr/lib/jvm/java-17-openjdk/bin/java --add-opens=java.base/java.util=ALL-UNNAMED --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.lang.invoke=ALL-UNNAMED --add-opens=java.prefs/java.util.prefs=ALL-UNNAMED --add-opens=java.base/java.nio.charset=ALL-UNNAMED --add-opens=java.base/java.net=ALL-UNNAMED --add-opens=java.base/java.util.concurrent.atomic=ALL-UNNAMED -XX:MaxMetaspaceSize=384m -XX:+HeapDumpOnOutOfMemoryError -Xms256m -Xmx512m -Dfile.encoding=utf8 -Duser.country=US -Duser.language=en -Duser.variant -cp /home/hasezoey/.gradle/wrapper/dists/gradle-8.6-bin/afr5mpiioh2wthjmwnkmdsd5w/gradle-8.6/lib/gradle-launcher-8.6.jar -javaagent:/home/hasezoey/.gradle/wrapper/dists/gradle-8.6-bin/afr5mpiioh2wthjmwnkmdsd5w/gradle-8.6/lib/agents/gradle-instrumentation-agent-8.6.jar org.gradle.launcher.daemon.bootstrap.GradleDaemon 8.6
hasezoey 2293224  0.8  2.0 9803420 329388 ?      Ssl  17:20   0:44 /usr/lib/jvm/java-17-openjdk/bin/java --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.base/java.lang=ALL-UNNAMED --add-opens java.base/java.lang.invoke=ALL-UNNAMED --add-opens java.base/java.util=ALL-UNNAMED --add-opens java.prefs/java.util.prefs=ALL-UNNAMED --add-opens java.prefs/java.util.prefs=ALL-UNNAMED --add-opens java.base/java.nio.charset=ALL-UNNAMED --add-opens java.base/java.net=ALL-UNNAMED --add-opens java.base/java.util.concurrent.atomic=ALL-UNNAMED -XX:MaxMetaspaceSize=256m -XX:+HeapDumpOnOutOfMemoryError -Xms256m -Xmx512m -Dfile.encoding=UTF-8 -Duser.country=US -Duser.language=en -Duser.variant -cp /home/hasezoey/.gradle/wrapper/dists/gradle-7.5-bin/f1w0cc9nuo0eivz97x2xlu9sv/gradle-7.5/lib/gradle-launcher-7.5.jar org.gradle.launcher.daemon.bootstrap.GradleDaemon 7.5
hasezoey 2299305  0.7  1.5 4410000 256996 ?      Sl   17:36   0:32 /usr/lib/jvm/java-17-openjdk/bin/java -cp /home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-compiler-embeddable/1.9.23/460f3eeb4ac5383e60a09898bafa9b63838cc2ea/kotlin-compiler-embeddable-1.9.23.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib/1.9.23/dbaadea1f5e68f790d242a91a38355a83ec38747/kotlin-stdlib-1.9.23.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-script-runtime/1.9.23/1375a7343d05891119fac57f8b50a7578420ac27/kotlin-script-runtime-1.9.23.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-reflect/1.6.10/1cbe9c92c12a94eea200d23c2bbaedaf3daf5132/kotlin-reflect-1.6.10.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-daemon-embeddable/1.9.23/a6ad5e888015dc4f9eb828ea753167f256093c6a/kotlin-daemon-embeddable-1.9.23.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.intellij.deps/trove4j/1.0.20200330/3afb14d5f9ceb459d724e907a21145e8ff394f02/trove4j-1.0.20200330.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains/annotations/13.0/919f0dfe192fb4e063e7dacadee7f8bb9a2672a9/annotations-13.0.jar -Djava.awt.headless=true -D$java.rmi.server.hostname=127.0.0.1 -Xmx512m -XX:MaxMetaspaceSize=384m -Dkotlin.environment.keepalive -ea --add-exports java.base/sun.nio.ch=ALL-UNNAMED org.jetbrains.kotlin.daemon.KotlinCompileDaemon --daemon-runFilesPath /home/hasezoey/.local/share/kotlin/daemon --daemon-autoshutdownIdleSeconds=7200 --daemon-compilerClasspath /home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-compiler-embeddable/1.9.23/460f3eeb4ac5383e60a09898bafa9b63838cc2ea/kotlin-compiler-embeddable-1.9.23.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-stdlib/1.9.23/dbaadea1f5e68f790d242a91a38355a83ec38747/kotlin-stdlib-1.9.23.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-script-runtime/1.9.23/1375a7343d05891119fac57f8b50a7578420ac27/kotlin-script-runtime-1.9.23.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-reflect/1.6.10/1cbe9c92c12a94eea200d23c2bbaedaf3daf5132/kotlin-reflect-1.6.10.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-daemon-embeddable/1.9.23/a6ad5e888015dc4f9eb828ea753167f256093c6a/kotlin-daemon-embeddable-1.9.23.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains.intellij.deps/trove4j/1.0.20200330/3afb14d5f9ceb459d724e907a21145e8ff394f02/trove4j-1.0.20200330.jar:/home/hasezoey/.gradle/caches/modules-2/files-2.1/org.jetbrains/annotations/13.0/919f0dfe192fb4e063e7dacadee7f8bb9a2672a9/annotations-13.0.jar

```

</details>

Thanks for helping debug this, for a while i though the increased RAM was once again firefox (previously i only grepped for `gradle` not `java`), i have no clue how those processes became residual, but i guess they were interfering with the android build after the first bad commit.

TL;DR: residual old version gradle process was still active, and the changed gradle version requirement set-off this misleading error

**Reply by benoit-pierre** [reply]
Ah! gradle launches a daemon, so subsequent compilations are faster.
